### PR TITLE
win-capture: Add and tweak apps in compatibility.json

### PR DESCRIPTION
**Pull request by TianQiBuTian**
### Description

Add:
Counter-Strike 2, UWP applications, Gaming Services applications, Microsoft 365(Excel, PowerPoint, Word, Access, OneNote, Outlook, Publisher and 365 software), Adobe After Effects, Adobe Character Animator, Adobe Photoshop, Adobe Premiere Pro, Steam, Epic Games Launcher, Ubisoft Connect, Tencent GUI applications, WeChat and YY.

Tweak:
Change "Electron" name to "Chromium";
Change "%name% cannot be captured via Game Capture." to "%name% cannot be captured using Game Capture.".

Remove:
Chrome and Edge (reason is based on Chromium).

Also add win-capture json in `.editorconfig`
### Motivation and Context

Before tweak: (Edge is not using `Compatibility.WindowCapture.BitBlt`)
QQ截图20230402055155

### How Has This Been Tested?

Local build and use 29.1.0-beta1 game capture.
### Types of changes

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: plugins/win-capture/window-capture.c
Comment (derrod) [inline plugins/win-capture/window-capture.c:1]: Please split this change out into another commit since it modifies the auomatic WGC list rather than compatibility data.

File: plugins/win-capture/data/locale/en-US.ini
Comment (derrod) [inline plugins/win-capture/data/locale/en-US.ini:54]: This may be misleading, UWP games can be game captured, but I'm not sure if they have a different window class.

File: plugins/win-capture/data/locale/en-US.ini
Comment (derrod) [inline plugins/win-capture/data/locale/en-US.ini:58]: You could use the generic translation strings for those, they do not need special ones.

**Review by derrod — CHANGES_REQUESTED** [review]
Mostly fine, though I do question the need for some of these.

Generally we'd want to keep this list to apps where users often run into issues. I'm not sure how often items like the Tencent apps come up, and I don't think many people want to 
specifically capture e.g. Steam or the Epic Launcher.

Others like PowerPoint and the generic UWP ones are certainly valid however.

Finally, you also need to increment the version in `package.json`.

**Comment by TianQiBuTian** [comment]
OK, I tested UWP games before submitting pr and couldn't use game capture. After downloading the OBS again, you can capture it. My bad.

The Tencent GUI is used on [QQ](https://im.qq.com/) and other Tencent software. Confirm that most software cannot use BitBlt.

And confirm that I need to add a version number? I checked https://obsproject.com/obs2_update/win-capture/v1/package.json status is still not found. And this feature is not currently available in the release version.

**Comment by derrod** [comment]
If you bump the version number now then we can push it independently of the next beta release and make sure before release that pulling the compatibility data updates works as expected.

File: plugins/win-capture/data/compatibility.json
Comment (derrod) [inline plugins/win-capture/data/compatibility.json:21]: Does this actually affect Chromium? Either way I'd prefer if you kept the existing translation key and just updated the text to say "Electron/Chromium" instead.

**Review by derrod — CHANGES_REQUESTED** [review]
Overall looks fine now, just one more thing.

File: plugins/win-capture/data/compatibility.json
Comment (TianQiBuTian) [reply plugins/win-capture/data/compatibility.json:21]: `Electron` includes `Chromium`. `Edge` and `Chrome` are also recognized as `Electron`, but they are not built on `Electron`.
https://wikipedia.org/wiki/Electron_(software_framework)

That's why I changed `Electron` to `Chromium`.

Like Steam, it can actually capture frames, but cannot capture pages. Because their sub windows use `Chromium`.
QQ截图20230402213709

**Comment by norihiro** [comment]
Please fold the commit descriptions at 72-column.

File: plugins/win-capture/data/compatibility.json
Comment (derrod) [reply plugins/win-capture/data/compatibility.json:21]: While Electron is using Chromium, the average user is more familiar with Electron than Chromium. Renaming it to "Electron/Chromium" and retaining the current translation key is probably the best way forward for clarity and translation workflow.

My question was more about whether Chromium/CEF based applications even use the `Chrome_WidgetWin_0` window class.

**Review by derrod — APPROVED** [review]
Looks fine to me now, I don't have a Windows machine handy right now so if somebody could confirm that Chrome/Electron still get correctly identified based on window class that would be great.

**Comment by TianQiBuTian** [comment]
I feel a bit strange about displaying Electron in front of only based on Chromium software. Can I swap Electron and Chromium tomorrow?

File: plugins/win-capture/data/locale/en-US.ini
Comment (derrod) [inline plugins/win-capture/data/locale/en-US.ini:54]: Actually the grammar here is wrong, it should either be "Applications based on Electron/Chromium" or "Electron/Chromium-based applications".

**Comment by derrod** [comment]
> I feel a bit strange about displaying Electron in front of only based on Chromium software. Can I swap Electron and Chromium tomorrow?

These filters are more likely to be displayed for Electron applications because those are simply more common and we already have an automatic switch to WGC for chrome/edge anyway. But perhaps @Warchamp7 has an opinion here.

**Comment by TianQiBuTian** [comment]
Idea: Can I remove specific message and change to `Games built on %name%`, `Applications based on %name%`?
Perhaps other engines will use this message in the future, so there is no need to add a special message. And there is no need to change the translation key when changing the name.

**Review by derrod — APPROVED** [review]
I think this is now mostly fine as-is, would love to get a second opinion from @Warchamp7 on the wording.

**Comment by Warchamp7** [comment]
My only feedback here is to simplify the Chromium/Electron warning to just 'Chromium'

A user is more likely to understand what Chromium means in the context of 'browser-based', and I think anyone who knows what the word Electron means in this context, will understand Chromium as a broader catch all

**Comment by TianQiBuTian** [comment]
Okay, this should be the last change (CS2 has been released).
It has been a long time since the first approval, so please review the code I changed again to see if there is any problem.

**Review by derrod — APPROVED** [review]

**Review by Warchamp7 — APPROVED** [review]

**Pull request merged**
